app/grandchallenge/components/__init__.py

```diff
@@ -23,3 +23,21 @@ def check_sagemaker_is_used(app_configs, **kwargs):
         )
 
     return errors
+
+
+@register(deploy=True)
+def check_dummy_provider_is_not_used(app_configs, **kwargs):
```

Comment:
I think it's also good to check INSTALLED_APPS in the same check. Also, a more logical place for this code might be profiles with the rest of the auth stuff or core.